paimon-core/src/main/java/org/apache/paimon/table/AbstractFileStoreTable.java

```diff
@@ -139,13 +138,13 @@ public RowKeyExtractor createRowKeyExtractor() {
 
     @Override
     public SnapshotReader newSnapshotReader() {
-        return newSnapshotReader(DEFAULT_MAIN_BRANCH);
+        return newSnapshotReader(CoreOptions.branch(options()));
     }
 
     @Override
     public SnapshotReader newSnapshotReader(String branchName) {
```

Comment:
Seems the branchName here is not used.

Reply:
> Seems the branchName here is not used.

Yes, I will contribute a PR and delete it uniformly